pages/posts/the-architectural-scope-of-json-schema.md

```diff
@@ -0,0 +1,76 @@
+
+However, in general, a system of behaviors is somewhat abstract, and so it's not practically useful.  What we need is a realization of this system in code.  What we need is an implementation.
+
+## Making JSON Schema useful
+
+Before we can make JSON Schema useful, we need to ask who it should be useful for.  To answer _that_ we need to know why JSON Schema exists in the first place.
+
+Well, we visited that in the previous section: a schema evaluates an instance to ensure that the instance complies with all of the rules represented by the schema.  If the instance does comply with the rules in the schema, then we say it is "valid" against that schema.
+
+The reasons to ensure we have valid JSON data could be numerous: checking data before deserializing into programmatic models, checking form inputs before submission, etc.  **These are the needs of applications.**
+
+So _applications_ are the consumers of JSON Schema.
+
+But an application can't consume a specification without that specification being realized into code.  That's where implementations come in.
+
+An implementation of JSON Schema is an embodiment of the specification that is directly consumable by an application.
+
+## A nuance
+
+Something I've seen a lot of lately, and I think the source of some of the confusion that has arisen in my discussions, is that an application may be merely an executable wrapper around an implementation.  This tends to give the appearance that the application itself is the implementation, in which case the application would be subject to the requirements of the specification.  But I don't see it that way.  Even in these cases, there exists a distinction between the application and the implementation, even if that distinction is, in practice, really blurry.
+
+Applications tend to have three basic components:  an interface (UX or API), some business logic, and data persistence.  **All** applications have an interface.  However, the business logic and data persistence components are optional to the degree that you can have one or the other or both.  (An application with only a UX is generally not very useful.)
+
+An application may only provide an interface over data persistence (e.g. a Postgres web service), meaning that there's no need for any business logic.  Conversely, another application may provide a computational service (e.g. image processing) where there's no need to persist data.
+
+For the recent conversations I've had, this second scenario seems to be the case: an application is created that just evaluates instances against schemas.  But this doesn't mean that the application and implementation *are* the same thing.
+
+Within these applications the implementation, which for these applications _is_ the business logic, is still a separate component from the interface.  And it's important to recognize that JSON Schema as a specification can only cover the implementation part.
```

Comment:
I'm trying not to call anyone (or any implementation) out on this.

> Do you just mean validators with a UI slapped on top?

That's the example that I'm using to illustrate the minimal case, but this could apply to any implementation.

> a JSON Schema implementation would always be a library?

I recognize that it might not be, but I don't know how to isolate the application from the requirements of the spec without making that implication.